File: app/src/main/java/evan/org/vienhance/main/MainActivity.java
package evan.org.vienhance.main;

/**
 * Create By yejiaquan in 2018/10/17 11:06
 */

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.*;
import evan.org.vienhance.Injection;
import evan.org.vienhance.R;
import evan.org.vienhance.domain.model.AlgContext;
import evan.org.vienhance.util.ActivityUtils;

import android.util.Log;
import evan.org.vienhance.util.AppExecutors;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "Main::Activity";

    private final AlgContext context;

    public MainActivity(){
        context = AlgContext.getInstance(new AppExecutors(), this);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        Log.i(TAG, "called onCreate");
        super.onCreate(savedInstanceState);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);

        setContentView(R.layout.main_act);

        MainFragment mainFragment = (MainFragment)getSupportFragmentManager().findFragmentById(R.id.contentFrame);

        if (mainFragment == null){
            mainFragment = MainFragment.newInstance();
            ActivityUtils.addFragmentToActivity(getSupportFragmentManager(),mainFragment,R.id.contentFrame);
        }

        MainPresenter presenter = new MainPresenter(mainFragment,
                Injection.provideUseCaseHandler(), context);
    }
}

File: app/src/main/java/evan/org/vienhance/main/MainActivity.kt
package evan.org.vienhance.main

import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import evan.org.vienhance.R
import evan.org.vienhance.util.ActivityUtils

import kotlinx.android.synthetic.main.main_act.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_act)
        setSupportActionBar(toolbar)

        var mainFragment: MainFragment? = supportFragmentManager.findFragmentById(R.id.contentFrame) as? MainFragment

        if (mainFragment == null) {
            mainFragment = MainFragment.newInstance()
            ActivityUtils.addFragmentToActivity(supportFragmentManager, mainFragment, R.id.contentFrame)
        }

        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }

        MainPresenter(mainFragment, applicationContext)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
